TypeSystem/Serialization/test.cc

```diff
@@ -115,6 +116,35 @@ CURRENT_STRUCT(WithTime) {
   CURRENT_FIELD(micros, std::chrono::microseconds, std::chrono::microseconds(0));
 };
 
+namespace named_variant {
+
+CURRENT_STRUCT(X) { CURRENT_FIELD(x, int32_t, 1); };
+
+CURRENT_STRUCT(Y) { CURRENT_FIELD(y, int32_t, 2); };
+
+CURRENT_STRUCT(Z) { CURRENT_FIELD(z, int32_t, 3); };
+
+CURRENT_STRUCT(T) { CURRENT_FIELD(t, int32_t, 4); };
+
+using A_VariantTypes = TypeListImpl<X, Y>;
+struct A_VariantName {
+  static const char* VariantNameImpl() { return "A"; }
+};
+using A = NamedVariant<A_VariantName, A_VariantTypes>;
+
+using B_VariantTypes = TypeListImpl<Z, T>;
+struct B_VariantName {
+  static const char* VariantNameImpl() { return "B"; }
+};
+using B = NamedVariant<B_VariantName, B_VariantTypes>;
+
+using E_VariantTypes = TypeListImpl<A, B>;
+struct E_VariantName {
+  static const char* VariantNameImpl() { return "E"; }
+};
+using E = NamedVariant<E_VariantName, E_VariantTypes>;
```

Comment:
Using `E` for `Variant` in this type test is a bit confusing - I was thinking about some `Enum` first :)